draft-ietf-webtrans-http3.md

```diff
@@ -571,6 +546,288 @@ A TLS exporter API might permit the context field to be omitted. In this case,
+end-to-end signal, intermediaries MUST consume flow control signals and express
+their own flow control limits to the next hop.  The intermediary can send these
+signals via HTTP/3 flow control messages, HTTP/2 flow control messages, or as
+WebTransport flow control capsules, where appropriate.  Intermediaries are
+responsible for storing any data for which they advertise flow control credit
+if that data cannot be immediately forwarded to the next hop.
+
+In practice, an intermediary that translates flow control signals between
+similar WebTransport protocols, such as between two HTTP/3 connections, can
+often simply reexpress the same limits received on one connection directly on
+the other connection.
+
+An intermediary that does not want to be responsible for storing data that
+cannot be immediately sent on its translated connection can ensure that it
+does not advertise a higher flow control limit on one connection than the
+corresponding limit on the translated connection.
+
+## Flow Control SETTINGS
+
+*[SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_UNI]: #
+*[SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_BIDI]: #
+*[SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA]: #
+
+Initial flow control limits can be exchanged via HTTP/3 SETTINGS
+({{http3-settings}}) by providing non-zero values for
+
+* WT_MAX_STREAMS via SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_UNI and
+  SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_BIDI
+* WT_MAX_DATA via SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA
+
+
+## Flow Control Capsules
+
+### WT_MAX_STREAMS Capsule {#WT_MAX_STREAMS}
+
+*[WT_MAX_STREAMS]: #
+
+An HTTP capsule {{HTTP-DATAGRAM}} called WT_MAX_STREAMS is introduced to inform
+the peer of the cumulative number of streams of a given type it is permitted to
+open.  A WT_MAX_STREAMS capsule with a type of 0x190B4D3F applies to
+bidirectional streams, and a WT_MAX_STREAMS capsule with a type of 0x190B4D40
+applies to unidirectional streams.
+
+Note that, because Maximum Streams is a cumulative value representing the total
+allowed number of streams, including previously closed streams, endpoints
+repeatedly send new WT_MAX_STREAMS capsules with increasing Maximum Streams
+values as streams are opened.
+
+~~~
+WT_MAX_STREAMS Capsule {
+  Type (i) = 0x190B4D3F..0x190B4D40,
+  Length (i),
+  Maximum Streams (i),
+}
+~~~
+{: #fig-wt_max_streams title="WT_MAX_STREAMS Capsule Format"}
+
+WT_MAX_STREAMS capsules contain the following field:
+
+   Maximum Streams:
+   : A count of the cumulative number of streams of the corresponding type that
+     can be opened over the lifetime of the connection. This value cannot
+     exceed 2<sup>60</sup>, as it is not possible to encode stream IDs larger
+     than 2<sup>62</sup>-1.
+
+An endpoint MUST NOT open more streams than permitted by the current stream
+limit set by its peer.  For instance, a server that receives a unidirectional
+stream limit of 3 is permitted to open streams 3, 7, and 11, but not stream
```

Comment:
Just talking out loud to remind myself - 

WT doesn't need logic about "implicitly" opening streams that have not arrived with lower stream numbers, like QUIC does, because there's no guarantee those quic streams are bound for this session.  A receiver just maintains a counter of how many streams have been opened and if that counter ever exceeds the last WT_MAX_STREAMS value then it's an error.